CanMatchPreFilterSearchPhaseTests > testSortShards FAILED

### CI Link

https://gradle-enterprise.elastic.co/s/75i35dy3fsa7e/tests/task/:server:test/details/org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests/testSortShards?top-execution=1

### Repro line

./gradlew ":server:test" --tests "org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests.testSortShards" -Dtests.seed=C49998B0F87552D6 -Dtests.locale=pl-PL -Dtests.timezone=Europe/Copenhagen -Druntime.java=22	

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

CanMatchPreFilterSearchPhaseTests > testSortShards FAILED
    java.lang.AssertionError: expected:<[logs][3]> but was:<[logs][5]>
        at __randomizedtesting.SeedInfo.seed([C49998B0F87552D6:A20BF86ABFFC582C]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests.testSortShards(CanMatchPreFilterSearchPhaseTests.java:368)

--- a/server/src/main/java/org/elasticsearch/search/sort/MinAndMax.java
+++ b/server/src/main/java/org/elasticsearch/search/sort/MinAndMax.java
@@ -60,7 +60,7 @@ public class MinAndMax<T extends Comparable<? super T>> implements Writeable {
         if (left == null) {
             return right == null ? 0 : -1; // nulls last
         }
-        return right == null ? -1 : left.getMin().compareTo(right.getMin());
+        return right == null ? 1 : left.getMin().compareTo(right.getMin());
     };
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
@@ -68,7 +68,7 @@ public class MinAndMax<T extends Comparable<? super T>> implements Writeable {
         if (left == null) {
             return right == null ? 0 : 1; // nulls first
         }
-        return right == null ? 1 : right.getMax().compareTo(left.getMax());
+        return right == null ? -1 : right.getMax().compareTo(left.getMax());
     };
 
     /**

